### PR TITLE
adding seven segment using decoder impelementation

### PR DESCRIPTION
**Pull request by noran97**

### CODE REVIEW

**Comment by AbdelrahmanBadr22** [comment]
Hi @noran97 ,
Based on the new changes, it looks like this PR has a conflict issue. Could you please resolve it and let me know?

File: C/SEGMENT_DECODER.c [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#include "SEGMENT_DECODER.h"
+#include "Types.h"
+
+error_t SEGMENT_DECODER_INIT(SEGMENT_DECODER * seven_segment)
```
Comment (AbdelrahmanBadr22) [inline C/SEGMENT_DECODER.c:4]: For pointers we use `p` so in this case we can use `pSevenSeg`

File: C/SEGMENT_DECODER.c [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+{
+    error_t Ret_ErrorState = kNoError ;
+	/* Make sure that the seven_segment is  valid */
+	if(seven_segment!= NULL) 
+	{
+        GPIO_SetPinDirection(seven_segment->segment[0].port_name,seven_segment->segment[0].pin_name,kOutput);
+        GPIO_SetPinDirection(seven_segment->segment[1].port_name,seven_segment->segment[1].pin_name,kOutput);
+        GPIO_SetPinDirection(seven_segment->segment[2].port_name,seven_segment->segment[2].pin_name,kOutput);
+        GPIO_SetPinDirection(seven_segment->segment[3].port_name,seven_segment->segment[3].pin_name,kOutput);
```
Comment (AbdelrahmanBadr22) [inline C/SEGMENT_DECODER.c:13]: What about using for loop : 
e.g. 
```
for (i = 0 ; i < TOTAL_DECODER_PINS ; i++) {
    GPIO_SetPinDirection(seven_segment->segment[i].portNumber,seven_segment->segment[i].pinNumber,kOutput);
}
```

File: C/SEGMENT_DECODER.h [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#ifndef LCD_H_
```
Comment (AbdelrahmanBadr22) [inline C/SEGMENT_DECODER.h:1]: Use correct macro name

**Review by AbdelrahmanBadr22 — CHANGES_REQUESTED** [review]

File: C/SEVEN_SEGMENT.c [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    else
+    {
+        error_t Ret_ErrorState = kFunctioParameterError ;
+    }
+    return Ret_ErrorState;
+
+}
+void SetNumber(sevenSegment_t * pSevenSeg,uint8_t number)
+{
+    int i=0;
```
Comment (AbdelrahmanBadr22) [inline C/SEVEN_SEGMENT.c:60]: Don't use `int`

**Review by AbdelrahmanBadr22 — CHANGES_REQUESTED** [review]

**Comment by noran97** [comment]
@AbdelrahmanBadr22 
Can you please kindly review my code 💐

**Comment by AbdelrahmanBadr22** [comment]
Hi @noran97 ,
There are a tricky bug with your code, remember if you didn't the new directory to the `build.dir` then the directory won't be built then any compiler errors in the new directory won't be detected and one can use this files. In the future we will automate detecting new directory and avoid using `build.dir` but for now you have to add the new directories to the `build.dir`

**Review by AbdelrahmanBadr22 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**